L2SIVacuum/POUs/Functions/Valves/FB_VCN_VAT590.TcPOU

```diff
@@ -278,7 +441,168 @@ stVcnVat590.stValveStatus.bPositionSPOutOfRange := i_stExtnWarnings.Bit6Position
+      </Implementation>
+    </Method>
+    <Action Name="WriteCoeParameters" Id="{9f7f239a-46a5-4609-b528-6b15a6fa6552}">
+      <Implementation>
+        <ST><![CDATA[// User will change one parameter at a time so solution with ELSIF should work OK
+IF stVcnVat590.bCoeError THEN
+	IF rtCoeReset.Q THEN
+		stVcnVat590.bCoeError := FALSE;
+	END_IF;
+ELSIF bFirstRead THEN
+	;//Wait until first readout of all CoE parameters is completed
+ELSIF stVcnVat590.stCoeParameters.ePIDCtrl <> stCoeParamsInternal.ePIDCtrl THEN
+	fbCoeWrite.nIndex := cPressCtrl.nIndex;
+	fbCoeWrite.nSubIndex := cPressCtrl.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.ePIDCtrl);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.ePIDCtrl);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.fSensorDelay_AD <> stCoeParamsInternal.fSensorDelay_AD THEN
+	fbCoeWrite.nIndex := cSensorDelay_AD.nIndex;
+	fbCoeWrite.nSubIndex := cSensorDelay_AD.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.fSensorDelay_AD);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.fSensorDelay_AD);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.fRampTime_AD <> stCoeParamsInternal.fRampTime_AD THEN
+	fbCoeWrite.nIndex := cRampTime_AD.nIndex;
+	fbCoeWrite.nSubIndex := cRampTime_AD.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.fRampTime_AD);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.fRampTime_AD);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.bRampMode_AD <> stCoeParamsInternal.bRampMode_AD THEN
+	fbCoeWrite.nIndex := cRampMode_AD.nIndex;
+	fbCoeWrite.nSubIndex := cRampMode_AD.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.bRampMode_AD);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.bRampMode_AD);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.fGainFactor_AD <> stCoeParamsInternal.fGainFactor_AD THEN
+	fbCoeWrite.nIndex := cGainFactor_AD.nIndex;
+	fbCoeWrite.nSubIndex := cGainFactor_AD.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.fGainFactor_AD);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.fGainFactor_AD);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.fRampTime_FIX1 <> stCoeParamsInternal.fRampTime_FIX1 THEN
+	fbCoeWrite.nIndex := cRampTime_FIX1.nIndex;
+	fbCoeWrite.nSubIndex := cRampTime_FIX1.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.fRampTime_FIX1);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.fRampTime_FIX1);
+	fbCoeWrite.bExecute := TRUE;
+ELSIF stVcnVat590.stCoeParameters.bRampMode_FIX1 <> stCoeParamsInternal.bRampMode_FIX1 THEN
+	fbCoeWrite.nIndex := cRampMode_FIX1.nIndex;
+	fbCoeWrite.nSubIndex := cRampMode_FIX1.nSubIndex;
+	fbCoeWrite.pSrcBuf := ADR(stVcnVat590.stCoeParameters.bRampMode_FIX1);
+	fbCoeWrite.cbBufLen := SIZEOF(stVcnVat590.stCoeParameters.bRampMode_FIX1);
+	fbCoeWrite.bExecute := TRUE;
```

Comment:
One small question: do you know that all of these parameters can be updated live?
There were a few CoE parameters in the stepper motor terminals that didn't take effect until the terminal was taken out of and back into the OP state.

Reply:
Yes, they can be updated live. I was able to test that.